### PR TITLE
Tickets/dm 38695

### PR DESCRIPTION
**Pull request by eacharles**

### CODE REVIEW

**Review by fritzm — APPROVED** [review]
LGTM

File: src/lsst/cm/tools/db/sqlalch_interface.py [hunk 1]
```diff
@@ -688,3 +702,20 @@ def _get_id(self, level: LevelEnum, parent_id: Optional[int], match_name: Option
...
         if entry is None:  # pragma: no cover
             raise ValueError(f"Failed to get entry for {db_id} at {level.name}")
+
+    def _check_terminal_state(self, level: LevelEnum, db_id: DbId) -> bool:
+        entry = self.get_entry(level, db_id)
+        terminal_states = list[
+            StatusEnum.failed,
+            StatusEnum.rejected,
+            StatusEnum.reviewable,
+            StatusEnum.rescueable,
```
Comment (eigerx) [inline src/lsst/cm/tools/db/sqlalch_interface.py]: This is missing superseded!

Comment (eigerx) [reply src/lsst/cm/tools/db/sqlalch_interface.py]: 🥉

**Review by eigerx — CHANGES_REQUESTED** [review]

**Review by eigerx — APPROVED** [review]
congratulations!

**Pull request merged**